### ISSUE TITLE
syntax_check: Component "fieldname" not found in structure (READ TABLE WITH KEY)

### ISSUE THREAD

**Issue by ed-holland** [abaplint/abaplint#2340]
node version:     v16.13.1
abaplint version: 2.83.22

Messages:
```
./src/nw/#jcs#bc.fugr.#jcs#lbcf90.abap[61, 3]          - Component "fieldname" not found in structure (check_syntax) [E]
./src/nw/#jcs#bc.fugr.#jcs#lbcf90.abap[61, 3]          - Component "fieldname" not found in structure (check_syntax) [E]
./src/nw/#jcs#bc.fugr.#jcs#lbcf90.abap[61, 3]          - Component "fieldname" not found in structure (check_syntax) [E]
./src/nw/#jcs#bc.fugr.#jcs#lbcf90.abap[61, 3]          - Component "fieldname" not found in structure (check_syntax) [E]
```

Code:
![image](https://user-images.githubusercontent.com/53995991/149474236-1e60eff5-a2a4-47d2-9cd6-ccf2e62b93a0.png)

**Comment by ed-holland** [comment]
2nd Case:

Message: `./src/fcc/#jcs#fcc.fugr.#jcs#lfccf01.abap[175, 7]      - Component "fieldname" not found in structure (check_syntax) [E]`

Code:
![image](https://user-images.githubusercontent.com/53995991/149474680-b7d29345-9494-45b9-b516-c04a3d7031fd.png)

**Comment by ed-holland** [comment]
I've updated the image with the complete form code

**Comment by larshp** [comment]
is `X031L` a known object in the repository?

**Comment by ed-holland** [comment]
Yes, it is part of the dependency repository

![image](https://user-images.githubusercontent.com/53995991/149475516-68574c49-25aa-4833-8f6f-5ddf10c19196.png)

The fieldname field is in the Include Structure
![image](https://user-images.githubusercontent.com/53995991/149475717-ff1bc5c9-1fa6-4aba-b8bc-d2da6f2443cf.png)

which is also in the deps repo
![image](https://user-images.githubusercontent.com/53995991/149475839-d32fe763-16ab-43b2-bd2a-3fba9b8d6d2e.png)

**Comment by larshp** [comment]
did the errors change, before and after?

**Comment by ed-holland** [comment]
The messages from the daily lint check are:
![image](https://user-images.githubusercontent.com/53995991/149730969-daee029c-b1ea-4675-bcda-9dcd60c68513.png)

From this one can see that the 2nd message is from a different include. Same error.
![image](https://user-images.githubusercontent.com/53995991/149743304-4333eb14-3922-45a7-b17f-4c20b91635ea.png)

The SCI in the above was just the FUGR. I've reran it for the package and got these messages. It is not consistent with the linter result.
![image](https://user-images.githubusercontent.com/53995991/149742782-bdf308e0-37b0-4273-8b17-688cab93870c.png)

**Comment by ed-holland** [comment]
Any idea where this comes from?

**Comment by larshp** [comment]
no, need more details or a reproduceable example

**Comment by ed-holland** [comment]
I could reproduce one case using the playground...

```
REPORT zfoobar.

TYPES: BEGIN OF incl,
  fieldname(30),
  END OF incl.

DATA: BEGIN OF x031l.
DATA: INCLUDE TYPE incl.
DATA: END OF x031l.

FORM check_table_field USING i_table i_field CHANGING l_rc o_return.

  DATA: tb_fields LIKE x031l OCCURS 0 WITH HEADER LINE.
*
* check if field exists
*
  READ TABLE tb_fields WITH KEY fieldname = i_field.
  IF sy-subrc NE 0.
    l_rc = 1.
  ENDIF.

ENDFORM.                    "check_table

FORM get_table_fields TABLES l_tab USING i_table.

  DATA: tb_fields TYPE x031l OCCURS 0 WITH HEADER LINE.
*
* append fields to table
*
  LOOP AT tb_fields.
    l_tab = tb_fields-fieldname.
    APPEND l_tab.
  ENDLOOP.

ENDFORM.                    "get_table_field
```

![image](https://user-images.githubusercontent.com/53995991/150535914-64480a4c-8481-4220-83e2-1294f068ed34.png)

**Comment by larshp** [comment]
thanks, in the above example, I think the finding is correct

![image](https://user-images.githubusercontent.com/5888506/150629906-ddf327d8-01ce-4bbb-87a7-84b783404d5b.png)

**Comment by ed-holland** [comment]
You are right, the DATA needs to go. A local structure Type does not have the same issue as a dictionary type. So playground can not help here.

The issues comes from the structure in the dependency repo.

```
REPORT zfoobar.

FORM check_table_field USING i_table i_field CHANGING l_rc o_return.

  DATA: tb_fields LIKE x031l OCCURS 0 WITH HEADER LINE.
*
* check if field exists
*
  READ TABLE tb_fields WITH KEY fieldname = i_field.
  IF sy-subrc NE 0.
    l_rc = 1.
  ENDIF.

ENDFORM.                    "check_table

FORM get_table_fields TABLES l_tab USING i_table.

  DATA: tb_fields TYPE x031l OCCURS 0 WITH HEADER LINE.
*
* append fields to table
*
  LOOP AT tb_fields.
    l_tab = tb_fields-fieldname.
    APPEND l_tab.
  ENDLOOP.

ENDFORM.                    "get_table_field
```

This should on a nw only system.

**Comment by ed-holland** [comment]
Issue is resolved with 2.85.12

Thanks

**Comment by larshp** [comment]
heh, hmm, ok
